internal/controllers/machinedeployment/machinedeployment_rollout_planner_test.go

```diff
@@ -499,10 +501,71 @@ func machineSetControllerMutator(log *fileLogger, ms *clusterv1.MachineSet, scop
+	// TODO(in-place): this should not be implemented in production code for the MachineSet controller
 	sortMachineSetMachines(scope.machineSetMachines[ms.Name])
 
+	// Removing updatingInPlaceAnnotation from machines after pendingAcknowledgeMove is gone in a previous reconcile (so inPlaceUpdating lasts one reconcile more)
+	// NOTE: This is a testing workaround to simulate inPlaceUpdating being completed; it is implemented in the fake MachineSet controller
+	// because while testing RolloutUpdate sequences with in-place updates we are not currently using a fake Machine controller.
+	// TODO(in-place): this should not be implemented in production code for the MachineSet controller nor in the production code for the Machine controller
+	replicasEndingInPlaceUpdate := sets.Set[string]{}
+	for _, m := range scope.machineSetMachines[ms.Name] {
+		if _, ok := m.Annotations[clusterv1.PendingAcknowledgeMoveAnnotation]; ok {
+			continue
+		}
+		if _, ok := m.Annotations[clusterv1.UpdateInProgressAnnotation]; ok {
+			delete(m.Annotations, clusterv1.UpdateInProgressAnnotation)
+			replicasEndingInPlaceUpdate.Insert(m.Name)
+		}
+	}
+	if replicasEndingInPlaceUpdate.Len() > 0 {
+		log.Logf("[MS controller] - Replicas %s completed in place update", sortAndJoin(replicasEndingInPlaceUpdate.UnsortedList()))
+	}
+
+	// If the MachineSet is accepting replicas from other MS (this is the newMS controlled by a MD),
+	// detect if there are replicas still pending AcknowledgeMove.
+	acknowledgeMoveReplicas := sets.Set[string]{}
+	if replicaNames, ok := ms.Annotations[clusterv1.AcknowledgedMoveAnnotation]; ok && replicaNames != "" {
+		acknowledgeMoveReplicas.Insert(strings.Split(replicaNames, ",")...)
+	}
+	notAcknowledgeMoveReplicas := sets.Set[string]{}
+	if sourceMSs, ok := ms.Annotations[clusterv1.MachineSetReceiveMachinesFromMachineSetsAnnotation]; ok && sourceMSs != "" {
+		for _, m := range scope.machineSetMachines[ms.Name] {
+			if _, ok := m.Annotations[clusterv1.PendingAcknowledgeMoveAnnotation]; !ok {
+				continue
+			}
+
+			// If machine has been acknowledged by the MachineDeployment, cleanup pending AcknowledgeMove annotation from the machine
+			if acknowledgeMoveReplicas.Has(m.Name) {
+				delete(m.Annotations, clusterv1.PendingAcknowledgeMoveAnnotation)
+				continue
+			}
+
+			// Otherwise keep track of replicas not yet acknowledged.
+			notAcknowledgeMoveReplicas.Insert(m.Name)
+		}
+	} else {
+		// Otherwise this MachineSet is not accepting replicas from other MS (this is an oldMS controller by a MD).
+		// Drop pendingAcknowledgeMoveAnnotationName from controlled Machines.
+		// Note: if there are machines recently moved but not yet accepted, those machines will be managed
+		// as any other machine and either moved to the new MS (after completing the in-place upgrade) or deleted.
+		for _, m := range scope.machineSetMachines[ms.Name] {
+			delete(m.Annotations, clusterv1.PendingAcknowledgeMoveAnnotation)
+		}
+	}
+
```

Comment:
```suggestion
```
Just to make it easier to figure out that this belongs to the block above (as it's a long func, this would be helpful for better structuring)